releasenotes/notes/control-flow-depth-size-b598a4eb9d8888eb.yaml

```diff
@@ -0,0 +1,15 @@
+---
+features:
+  - |
+    The circuit method :meth:`.QuantumCircuit.depth`, and the :class:`.DAGCircuit`
+    methods :meth:`~.DAGCircuit.depth` and :meth:`~.DAGCircuit.size` gained a
+    ``recurse`` keyword argument for use with control flow.  By default this is
+    ``False`` and will raise an error if control-flow is present, to avoid
+    nonsensical results.  If set to ``True``, a hint to the true value is returned,
```

Comment:
Does this imply that if `recurse=true`, we will return nonsensical results? :) Maybe "ambiguous" or "poorly defined"?

Reply:
Done in 61380dd9e